crates/primitives/Cargo.toml

```diff
@@ -22,22 +22,16 @@ rust_2018_idioms = "deny"
 all = "warn"
 
 [dependencies]
-alloy-primitives = { version = "0.8.2", default-features = false, features = [
+alloy-primitives = { version = "0.8.5", default-features = false, features = [
     "rlp",
+    "map",
 ] }
-hashbrown = "0.14"
-
-# utils
-cfg-if = { version = "1.0", default-features = false }
-
-[build-dependencies]
-hex = { version = "0.4", default-features = false }
 
 [features]
-default = ["std"]
-std = ["alloy-primitives/std", "hex/std"]
-hashbrown = []
-serde = ["alloy-primitives/serde", "hex/serde", "hashbrown/serde"]
+default = ["std", "alloy-primitives/default"]
```

Comment:
I would remove `alloy-primitives/default` to leave std map as a default. Other hashers could be enabled by alloy-primitives features

Reply:
Sure, do you want to expose these features in revm too?

Reply:
A small problem with this is that some methods only work with the std hasher (`new`, `with_capacity`, `From::from`...), which is fine if it's the default, but if you change hasher revm will fail to compile and it might not be caught in CI

Reply:
Was surprised by `new`, `From::from` when I looked at the changes. We should have CI that switches to non standard hasher.

I want to experiment with this a little, features get unified if **same** version of dependency is included, as in include both `revm` and `revm-primitives`, so we could by enabling revm-primitives features enable them inside revm.

Need to check what happens if versions are not same, and how brittle is this pattern.

You don't need to do it now in this PR, but I will maybe add it later when I come to this.